packages/core-v2-sdk/src/datasources/MultiTerm/MultiTermContractDataSource.ts

```diff
@@ -119,4 +121,72 @@ export class MultiTermContractDataSource
     const supply = await this.call("totalSupply", [termId]);
     return fromBn(supply, await this.getDecimals());
   }
+
+  /**
+   * Wraps the lock function in the Term contract, allows caller lto mint fixed and variable positions in a term.
+   * @async
+   * @param {Signer} signer - Ethers signer object.
+   * @param {string[]} assetIds -  The array of PT, YT and Unlocked share identifiers.
+   * @param {string[]} assetAmounts - The amount of each input PT, YT and Unlocked share to use
+   * @param {number} termId - The term id (expiry).
+   * @param {string} amount - Amount of underlying tokens to use to mint.
+   * @param {string} ptDestination - Address to receive principal tokens.
+   * @param {string} ytDestination - Address to receive yield tokens.
+   * @param {string} hasPreFunding- Have any funds already been sent to the contract, not commonly used for EOAs.
+   * @return {Promise<MintResponse>}
+   */
+  async lock(
+    signer: Signer,
+    termId: number,
+    assetIds: string[],
+    assetAmounts: string[],
+    amount: string,
+    ptDestination: string,
+    ytDestination: string,
+    ytBeginDate: number,
+    hasPreFunding: boolean,
+  ): Promise<MintResponse> {
+    if (assetAmounts.length !== assetIds.length) {
+      throw new Error(
+        "Error MultiTermDataSource.Lock(): assetIds and assetAmounts must be the same length.",
+      );
+    }
+
+    const assetIdsUnique = new Set(assetIds);
+    if (assetIdsUnique.size !== assetIds.length) {
+      throw new Error(
+        "Error MultiTermDataSource.Lock(): assetIds list is not unique.",
+      );
+    }
+
+    const multiTerm = this.contract.connect(signer);
+    const txn = await multiTerm.lock(
+      assetIds,
+      assetAmounts,
+      amount,
+      hasPreFunding,
+      ytDestination,
+      ptDestination,
+      ytBeginDate,
+      termId,
+    );
+    const receipt = await txn.wait();
+    // can safely assume that any successful transaction will have events and event arguments
+    const events = receipt.events!;
+    const transferSingleLogs = events.filter((event) => {
+      return (
+        event.event === "TransferSingle" &&
+        event.args!.from === ethers.constants.AddressZero
+      );
+    });
+
+    // maybe can just check index but not verified events will be ordered
+    const ytMintEvent = transferSingleLogs.find((log) => isYT(log.args!.id))!;
+    const ptMintEvent = transferSingleLogs.find((log) => isPT(log.args!.id))!;
```

Comment:
I'm not understanding the comment 🤔

Reply:
for any mint transaction there will be exactly two events, one minting and one for pt. So we could access the elements in the transferSingleLogs array by index but I have not verified that order of the events is guaranteed. so yt mint event always before PT. so using find is a bit safer alt